google-cloud-spanner/test/google/cloud/spanner/session/commit_test.rb

```diff
@@ -62,20 +71,40 @@
         )
       )
     ]
+  }
 
+  it "commits using a block" do
     mock = Minitest::Mock.new
-    mock.expect :commit, commit_resp, [{ session: session.path, mutations: mutations, transaction_id: nil, single_use_transaction: tx_opts }, default_options]
+    mock.expect :commit, commit_resp, [{ session: session.path, mutations: mutations, transaction_id: nil, single_use_transaction: tx_opts, return_commit_stats: nil }, default_options]
 
     session.service.mocked_service = mock
 
-    timestamp = session.commit do |c|
+    resp = session.commit do |c|
       c.update "users", [{ id: 1, name: "Charlie", active: false }]
       c.insert "users", [{ id: 2, name: "Harvey",  active: true }]
       c.upsert "users", [{ id: 3, name: "Marley",  active: false }]
       c.replace "users", [{ id: 4, name: "Henry",  active: true }]
       c.delete "users", [1, 2, 3, 4, 5]
     end
-    _(timestamp).must_equal commit_time
+    assert_commit_resp resp
+
+    mock.verify
+  end
+
+  it "commits using a block and return commit stats" do
```

Comment:
I feel that this adds a lot of redundant test code. Can we try to avoid these changes?

Reply:
Yes working on redundant test case code.